Pypy windows

mattip opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->
Continuation of #38, but I cannot push to that branch.

Adds some patches for PyPy7.3.5 prolbmes that will be fixed for PyPy7.3.6

But in the end, we come up against 
```
PythonService.cpp
win32/src/PythonService.cpp(622): error C3861: 'Py_SetProgramName': identifier not found
win32/src/PythonService.cpp(627): error C3861: 'Py_Initialize': identifier not found
win32/src/PythonService.cpp(639): error C3861: 'PySys_SetArgv': identifier not found
```

which is https://github.com/mhammond/pywin32/issues/1289, with a potential work-around in a [comment there](https://github.com/mhammond/pywin32/issues/1289#issuecomment-447177523)

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

mattip commented:
The [azure pipelines builds](https://dev.azure.com/conda-forge/feedstock-builds/_build?definitionId=6007&_a=summary) are not shown in the "checks" section.

carlodri commented:
> The [azure pipelines builds](https://dev.azure.com/conda-forge/feedstock-builds/_build?definitionId=6007&_a=summary) are not shown in the "checks" section.

I'm experiencing the same thing across all conda-forge stuff, @conda-forge/core any idea on what's happening?

mattip commented:
See https://github.com/conda-forge/status/issues/117 for the tracking issue. So far there is no further information.

carlodri commented:
@mattip thank you for the feedback!

isuruf commented:
@conda-forge-admin, rerender

mattip commented on recipe/patches/0001-remove-modules-needing-pythonservice.cpp-for-pypy.patch:
```diff
@@ -0,0 +1,74 @@
+     # source file base,     "base class" for generated COM support
+-- 
+2.28.0
+
```
I think it might be cleaner to do something like this
```
diff --git a/setup.py b/setup.py
index 297d15cb..40794da1 100644
--- a/setup.py
+++ b/setup.py
@@ -102,6 +102,8 @@ try:
 except KeyError:
     pass # no EXT_SUFFIX in this build.
 
+IS_PYPY = sys.implementation.name == "pypy"
+
 build_id_patch = build_id
 if not "." in build_id_patch:
     build_id_patch = build_id_patch + ".0"
@@ -635,6 +637,10 @@ class my_build_ext(build_ext):
         if ext.platforms and self.plat_name not in ext.platforms:
             return "Only available on platforms %s" % (ext.platforms,)
 
+        if IS_PYPY and ext.name in ('servicemanager', 'pythonservice',
+                                    'pythoncom', 'PyISAPI_loader'):
+            return "Cannot build service extensions on PyPy"
+
         # We update the .libraries list with the resolved library name.
         # This is really only so "_d" works.
         ext.libraries = patched_libs
```

isuruf replied:
Sounds good. Can you push your changes here?

mattip replied:
I am working out exactly which modules fail locally. So far I need to add 'adsi' to this list, maybe more.

mattip commented:
reopen to restart CI

mattip commented:
CI is passing. I have removed a number of modules on PyPy, I hope that downstream dependencies will not fail because of that.

Removed modules:
- `pythoncom`, and its dependents `adsi`, `axcontrol`, `axscript`, `axdebug`, `internet`, `mapi`, `exchange`, `shell`, `propsys`, `taskscheduler`, `bits`, `ifilter`, `directsound`, `authorization`
- `win32ui`, and its dependents `win32uiole`, `dde`
- `servicemanager`, `pythonservice`, and `PyISAPI_loader`

mattip commented:
In a comment on the parallel pywin32 issue, the project manager ask that we not merge this since it is missing capabilities and the project does not want to deal with bug reports. Closing.
